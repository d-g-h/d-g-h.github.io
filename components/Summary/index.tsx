type SummaryProps = {
  summary: string;
};

import styles from "@/components/Summary/summary.module.css";

export default function Summary({ summary }: Readonly<SummaryProps>) {
  return (
    <>
      <h2
        className={styles.title}
        style={{ textDecoration: "underline", marginTop: "0.75rem" }}
      >
        Summary
      </h2>
      <p className={styles.summary}>{summary}</p>
    </>
  );
}
