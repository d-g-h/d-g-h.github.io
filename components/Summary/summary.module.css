.title {
  font-size: 1.25rem;
  font-weight: bold;
}

.summary {
  font-size: 0.75rem;
}
