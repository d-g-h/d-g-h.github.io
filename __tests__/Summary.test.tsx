/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Summary from "@/components/Summary";

const SUMMARY = "Sum of the summary";

describe("Summary", () => {
  it("renders Summary", () => {
    render(<Summary summary={SUMMARY} />);

    const summary = screen.getByText(SUMMARY);

    expect(summary).toBeInTheDocument();
  });
  it("renders html correctly", () => {
    const { container } = render(<Summary summary={SUMMARY} />);
    expect(container).toMatchSnapshot();
  });
});
