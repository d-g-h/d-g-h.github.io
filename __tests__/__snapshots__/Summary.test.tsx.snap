// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary renders html correctly 1`] = `
<div>
  <h2
    class="title"
    style="text-decoration: underline; margin-top: 0.75rem;"
  >
    Summary
  </h2>
  <p
    class="summary"
  >
    Sum of the summary
  </p>
</div>
`;
